### ISSUE TITLE
upload file button

### ISSUE THREAD

**Issue by JK742020** [humhub/wiki#309]
Wouldn't it be nice to have a upload file button the same way like in posts or messages? It would be more visible than a link in the text only I think. Thanks

**Comment by marc-farre** [comment]
I would also allow having a video player instead of only a link to download the file.
See https://community.humhub.com/content/perma?id=276032

**Comment by WizWareTech** [comment]
Yes, this would REALLY help the Wiki module as a whole. Not only just video player, but same embedded audio player as well like it works for normal posts.

So if Wiki module could be adjusted to include ‘Attach’ button to input form, and then adjust the view to show attachments (just like they already show in the stream for stream posts), then Wiki's would work just the same.

**Comment by marc-farre** [comment]
@luke- Would you like me to send a PR for this issue?

It's about allowing to attach files to Wiki pages (same as Posts):

![Image](https://github.com/user-attachments/assets/38640f2d-1e95-4ab2-ba13-92b842f09ed6)

![Image](https://github.com/user-attachments/assets/f9e4e7ab-86b0-4e49-9812-0588dfcb58d9)

**Comment by luke-** [comment]
@marc-farre The problem here is that unfortunately files can then only be attached at the bottom of the page and are also difficult to edit. These are then often missed.

We would like to improve the positioning of embedded media in the future. 

Therefore, for UX reasons, I would not do this at the moment.

**Comment by WizWareTech** [comment]
@marc-farre, there is a new module that is coming that allows moving Posts to Wikis. Many times we find that users are posting something in the stream that should be better organized as a Wiki item. Prior to the 'Post to Wiki' module (coming soon), it required someone manually copying/pasting content from one to the other, and downloading any attachments from the stream post, then creating a comment on the Wiki post to then attach those attachments to. This put the attachments in an even worse place that was harder to see, in a comment below the bottom of the Wiki post.

Many times users want to attach inline videos/mp3 and the issue with uploading them using the toolbar option for inline on a post or wiki is that the inline attachments do not show an inline viewer for mp4 or mp3, and you can only get that if you use the attach feature.

So, in order for the Post to Wiki module to capture the content, attachments, comments, and likes, right now it would have to copy the content, likes, then create a new blank comment to the post in order to move the attachments to, then add the rest of the comments.

If it is fairly easy to mirror the attachment functionality in the Wiki as described above, then we aren't out anything, and the Wiki posts would behave consistently with the Stream posts, which would be intuitive, and the Post to Wiki module will work perfectly as it is and could be released as-is.

If not, then the Post to Wiki module will need to treat those attachments in an odd way.

And in the future of course, embedded players working with inline media would also be a really helpful thing, because often users want to attach an inline video to Wiki, but there is no way to do that with a player without attaching to a wiki comment.

Let me know your thoughts, and thank you, I'm really trying to help with constructive feedback after talking with users regularly and using the platform myself on a daily basis. I really think we haven't even begun to see the potential HumHub has to be a great platform for a multitude of different use-cases.
